client/src/www/app/outline_server_repository/vpn.ts

```diff
@@ -12,12 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-export interface ShadowsocksSessionConfig {
-  host?: string;
-  port?: number;
-  password?: string;
-  method?: string;
-  prefix?: string;
+import * as net from '@outline/infrastructure/net';
+
+// TunnelConfig represents the configuration to set up a tunnel.
+export interface TunnelConfig {
+  // transport describes how to establish connections to the destinations.
+  transport: TransportConfig;
+  // This is the place where routing configuration would go.
+}
+
+// TransportConfig describes how to establish connections to destinations.
+// We need to support a few meta operations such as getHost to support the UI.
+export class TransportConfig {
+  constructor(private readonly json: object) {}
+
+  // The address of the tunnel server, if there's a meaningful one.
+  // This is used to show the server address in the UI when connected.
+  getAddress(): string | undefined {
+    const hostConfig = this.json as {host?: string; port?: string};
```

Comment:
Can you use a type annotation here instead of a type assertion? That is generally safer but I'm not sure whether an assertion is required to cast.

```suggestion
    const hostConfig: {host?: string; port?: string} = this.json;
```

Reply:
Thanks for the link. I changed as per your suggestion.

I don't cast in the constructor because there will be transports that won't support getHost() (e.g. proxyless, multi-server, ...)